dashboard/app/controllers/admin_reports_controller.rb

```diff
@@ -0,0 +1,169 @@
+# The controller for reports of internal admin-only data.
+class AdminReportsController < ApplicationController
+  before_filter :authenticate_user!, except: [:header_stats]
+
+  before_action :set_script
+  include LevelSourceHintsHelper
+
+  def admin_concepts
+    authorize! :read, :reports
+    SeamlessDatabasePool.use_persistent_read_connection do
+      render 'admin_concepts', formats: [:html]
+    end
+  end
+
+  def funometer
+    authorize! :read, :reports
+
+    # Compute the global funometer percentage.
+    all_ratings = PuzzleRating.all
+    positive_ratings = all_ratings.where(rating: 1)
+    @global_percentage = 100.0 * positive_ratings.count / all_ratings.count
+
+    # Generate the funometer percentages, by day, for the last month.
+    percentages_by_day = all_ratings.select('100.0 * SUM(rating) / COUNT(rating)').where('created_at > ?', Time.now.prev_month).group('DATE(created_at)').order('DATE(created_at)').pluck('DATE(created_at)', '100.0 * SUM(rating) / COUNT(rating)')
```

Comment:
This also seems vulnerable to divide by zero, and more likely to get it given the filter.